nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/StandardProcessorNode.java

```diff
@@ -1321,12 +1325,21 @@ public void run() {
                         this.unscheduled = true;
                     }
                     try {
-                        if (activeThreadMonitorCallback.call()) {
+                        if (scheduleState.isScheduled()) {
+                            schedulingAgent.unschedule(StandardProcessorNode.this, scheduleState);
+                        }
```

Comment:
Fair point regarding the previously tighter encapsulation.  Given the closed nature of this model i think it comes down to a stylistic preference.  I'm ok with either for this case.